bbb-common-web/src/main/java/org/bigbluebutton/presentation/imp/PresentationProcessExternal.java

```diff
@@ -0,0 +1,185 @@
+package org.bigbluebutton.presentation.imp;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+
+public class PresentationProcessExternal {
+    private static Logger log = LoggerFactory.getLogger(PresentationProcessExternal.class);
+    private String presentationDir = "/var/bigbluebutton";
+    private long maxConversionTime = 5*60;
+    private boolean generatePngs = false;
+    private int thumbScale = 150; //used when converting pdf to thumb
+    private int maxImageWidth = 2048;
+    private int maxImageHeight = 1536;
+    private int placementsThreshold = 800;
+    private int imageTagThreshold = 800;
+    private int pngScaleTo = 800;
+    private int pngDefaultWidth = 500;
+    private int pngDefaultHeight = 500;
+    private int pdfFontsTimeout = 3;
+    private int maxNumberOfAttemptsForPdfFonts = 3;
+    private int svgPresentationResolutionPpi = 300;
+    private boolean forceRasterizeSlides = false;
+    private String pngWidthRasterizedSlides = "2048";
+    private String pageConversionMemoryHigh = "512M";
+    private String pageConversionMemoryMax = "640M";
+
+
+    public boolean processPage(String meetingId, String presentationId, int page) {
+        List<String> command = Arrays.asList(
+                "systemd-run", "--user", "--pipe", "--wait", "--same-dir",
+                "-p", "Environment=PRESENTATIONS_DIR=" + presentationDir,
+                "-p", "Environment=THUMBNAIL_SCALE=" + thumbScale,
+                "-p", "Environment=PDF_FONTS_TIMEOUT=" + pdfFontsTimeout,
+                "-p", "Environment=PDF_FONTS_MAX_ATTEMPTS=" + maxNumberOfAttemptsForPdfFonts,
+                "-p", "Environment=SVG_RESOLUTION_PPI=" + svgPresentationResolutionPpi,
+                "-p", "Environment=MAX_SVG_PATHS=" + placementsThreshold,
+                "-p", "Environment=MAX_SVG_IMGS=" + imageTagThreshold,
+                "-p", "Environment=MAX_IMAGE_WIDTH=" + maxImageWidth,
+                "-p", "Environment=MAX_IMAGE_HEIGHT=" + maxImageHeight,
+                "-p", "Environment=PNG_DEFAULT_WIDTH=" + pngDefaultWidth,
+                "-p", "Environment=PNG_DEFAULT_HEIGHT=" + pngDefaultHeight,
+                "-p", "Environment=PNG_SCALE_TO=" + pngScaleTo,
+                "-p", "Environment=GENERATE_PNGS=" + booleanToString(generatePngs),
+                "-p", "Environment=RASTERIZE_SLIDE_FORCE=" + booleanToString(forceRasterizeSlides),
+                "-p", "Environment=RASTERIZE_PNG_WIDTH=" + pngWidthRasterizedSlides,
+                "-p", "ProtectSystem=strict",
+                "-p", "ProtectHome=yes",
+                "-p", "PrivateTmp=true",
+                "-p", "PrivateUsers=true",
```

Comment:
```suggestion
                "-p", "PrivateUsers=true",
                "-p", "PrivateNetwork=true",
```

It might be desirable to run the process in an isolated network namespace so no network access is possible. Is this code path called by office document conversion as well? If this is the case network isolation is IMHO a must as office documents may trigger network access on opening.

Reply:
I tried it before but I get this warning `run-u118.service: PrivateNetwork=yes is configured, but network namespace setup failed, ignoring: Operation not permitted`.


When the script is launched as a user unit (`systemd-run --user`), systemd lacks the CAP_SYS_ADMIN privilege required to create a network namespace.
https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#PrivateNetwork=